### PR TITLE
Improve Search Algorithm

### PR DESCRIPTION
**Pull request by zachseidner1**
## Overview

Implements the mongoDB text search functionality over the `Fuse.js` library, improving efficiency and accuracy of search results. For details of the algorithm, see [$text](https://www.mongodb.com/docs/manual/reference/operator/query/text/#mongodb-query-op.-text). Please keep in mind that for these changes to work, **you must index the collections** with respect to the fields that should be queried in the search. I have already done this in the production database, so this PR is ready for deployment. However if you are testing this PR locally with local data or on the dev database it will not work if the collections are not indexed. To create a text index, see [Create a Text Index](https://www.mongodb.com/docs/manual/core/indexes/index-types/index-text/create-text-index/#std-label-create-text-index)

## Changes Made

- Updated search endpoints to use `$text` operator, and updated documentation accordingly.
- Updated `README.md` to more accurately reflect start procedure.
- Updated unit testing procedure to create indexes for collections before running the unit tests.

## Test Coverage

The search unit tests still pass. I also used GraphQL to test different search queries, and all of them gave good results that matched my query.

## Next Steps

- Make sure that all collections are properly indexed on production database even after their schema changes.

### CODE REVIEW

File: start.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# To start:
+`direnv allow` 
+
+`eval "$(direnv hook zsh)"`
```
[the remark this answers is not in the document]
Comment (zachseidner1) [reply start.md:4]: > 



> We can specify here that this file should be added into `~/.zshrc`, and also that this is only the case if the user is running ZSH.

I don't really know what you mean. I just did `brew install direnv` or something like that. Could you add the documentation here in a change suggestion? Thanks.

**Comment by zachseidner1** [comment]
@Archit404Error thanks for the feedback, super helpful! How did you find out you were able to create indexes in the model? Just curious because I didn't see it in the documentation that mongo db provides on creating indexes. Anyway, thanks again for the feedback and I think those commits should address all of your changes. Lmk if there's anything else I should update!

File: README.md [hunk 2]
```diff
@@ -25,10 +25,16 @@ and install dependencies with
 
 `npm install`
 
-To run the project, use
+We use `direnv` to keep track of environment variables. Install `direnv` to your machine.
+Then, to start the project, run the following terminal commands:
+
+`direnv allow` 
+
+`eval "$(direnv hook zsh)"`
```
Comment (Archit404Error) [inline README.md:33]:
```suggestion
In order to allow direnv to load environment variables, it needs to be able to hook into your shell. If using ZSH, for example, you should add the following line to the end of your `~/.zshrc` file:
`eval "$(direnv hook zsh)"`

You can find the full list of direnv hook commands [here](https://direnv.net/docs/hook.html)

Then, to start the project, run the following terminal commands
`direnv allow` 

```

**Review by Archit404Error — APPROVED** [review]
Good to go, take a look at my suggested change below to see what I meant in the earlier review about the README update. The `~/.zshrc` is a specific initialization file that is run when your ZSH shell is sourced, equivalent files exist for bash and other types of shells. Rather than trying to hook direnv into your shell manually every time you want to run this backend you can just add the command to the end of this file and it'll run automatically whenever you open your shell.

Also, regarding the index decorator, I found it by searching the [typegoose docs](https://typegoose.github.io/typegoose/). A key concept to keep in mind here is that the backend uses Typegoose, which is an _abstraction_ on top of Mongoose, which itself is an abstraction on top of the MongoDB driver.

So if you're looking at some functionality that Mongoose/Mongo supports, it's worth taking a look at Typegoose to see if it exposes some sort of decorator or method that allows you to express that functionality via native TypeScript.

**Comment by zachseidner1** [comment]
Thanks again for reviewing these changes again @Archit404Error . It looks like when I committed your README update, the PR needs to be approved again before I can merge it. Also really interesting to here more about the Volume tech stack, I was wondering what libraries were being used so that really helps. Thanks!

**Review by Archit404Error — APPROVED** [review]
Good to go

**Pull request merged**